addons/l10n_ch/models/account_journal.py

```diff
@@ -18,14 +18,15 @@ class AccountJournal(models.Model):
         ('ch', 'Switzerland')
     ], ondelete={'ch': lambda recs: recs.write({'invoice_reference_model': 'odoo'})})
 
-    @api.model
-    def create(self, vals):
-        rslt = super(AccountJournal, self).create(vals)
-
-        # The call to super() creates the related bank_account_id field
-        if 'l10n_ch_postal' in vals:
-            rslt.l10n_ch_postal = vals['l10n_ch_postal']
-        return rslt
+    @api.model_create_multi
+    def create(self, vals_list):
+        journals = super(AccountJournal, self).create(vals_list)
+
+        for journal, vals in zip(journals, vals_list):
```

Comment:
I'm gonna (re)verify this, but it's already done in multiple create's (base/res_partner, stock/stock_move_line, mail/mail_thread, ...) in the current codebase.

Reply:
Ok, @xmo-odoo tells me the order is maintained, never mind :+1: Then you could use it in coupon_program.py too no?

Reply:
Why ? I don't need the vals in coupon_program.py :thinking:

Reply:
> Edit: It's better to use the final values instead of the given value anyway (this way we correctly consider the default values, not specified in vals/vals_list).

But surely if that was the goal the original code would have done that, no? There's a bunch of places where the original code only processed explicit inputs, which now process both explicit and implicit, was the original just mistaken in every case?

Reply:
IMO, for the majority of the models, the devs are coding while considering their Form view flow, not considering any creation of records through other ways (custo's, context modifications, ...).  They do not consider creation through "code", imports, ... .  Furthermore, their knowledge of ´default´ behavior is sometimes restricted to the `default_get` called by the web-client.

Using the final records values in create post-processing should be the way to go (except for some specific case). It won't change the "default/generic" behavior, while improving the robustness of the implementation, considering other potential changes (other create overrides, `default` values, ...).